NineAnimator/Models/Anime Source/animeunity.it/AnimeUnity+Anime.swift

```diff
@@ -21,91 +21,75 @@ import Foundation
 import SwiftSoup
 
 extension NASourceAnimeUnity {
+    struct SearchResponseRecordsAnime: Codable {
+        var number: String = ""
+        var link: String = ""
+    }
+    struct  SearchResponseRecordsData: Codable {
+        var title_eng: String?
+        var episodes: [SearchResponseRecordsAnime]
+    }
     func anime(from link: AnimeLink) -> NineAnimatorPromise<Anime> {
         self.requestManager
             .request(url: link.link, handling: .browsing)
-            .responseString
+            .responseData
             .then {
                 responseContent -> Anime in
-                let bowl = try SwiftSoup.parse(responseContent)
-                var animeTitle = try bowl.select(".content p").text()
-                _ = try bowl.select("div.card-body p").compactMap {entry -> String in
-                    let trama = ""
-                    if try entry.text().contains("TITOLO") {
-                        let title = try entry.text()
-                        animeTitle = String(title.dropFirst(8))
-                    }
-                    return trama
-                }
-                let animeArtworkUrl = URL(
-                    string: try bowl.select(".cover>img").attr("src")
-                    ) ?? link.image
+                let data = responseContent
+                let utf8Text = String(data: data, encoding: .utf8) ?? String(decoding: data, as: UTF8.self)
+                let bowl = try SwiftSoup.parse(utf8Text)
+                var encoded = try bowl.select("video-player").attr("anime")
+                encoded = encoded.replacingOccurrences(of: "\n", with: "")
+                let data_json = encoded.data(using: .utf8)!
+                let decoder = JSONDecoder.init()
```

Comment:
```suggestion
                let decoder = JSONDecoder()
```
SwiftLint or the swift compiler should give you a warning for this. Try not to use `Type.init()` for instantiation.